packages/shorebird_cli/lib/src/executables/aot_tools.dart

```diff
@@ -67,6 +67,39 @@ const _preLinkerFlutterRevisions = <String>{
   '1a6115bebe31e63508c312d14e69e973e1a59dbf',
 };
 
+/// {@template aot_tools_execution_failure}
+/// Exception thrown when aot_tools execution exits with a non-zero exit code.
+/// {@endtemplate}
+class AotToolsExecutionFailure implements Exception {
```

Comment:
This has really little to do with aottools, but it's fine to start with something specific for now and make it generic later.

Reply:
Yup happy to look at the rest of our wrappers and see if we can unify things a bit more in a subsequent PR 👍